model/protecode.py

```diff
@@ -35,4 +41,34 @@ def tls_verify(self):
 
 
 class ProtecodeCredentials(BasicCredentials):
-    pass
+    def auth_scheme(self):
```

Comment:
nit: return-type-hint. also, the base-class is a bit misleading now, is it not?
-> rather define two credentials-classes (one for basic-auth, one for token-auth), check for configured auth-type, and return the appropriate class, rather than mixing those two together